Java/core/src/main/java/com/amazon/randomcutforest/sampler/SimpleStreamSampler.java

```diff
@@ -202,7 +214,21 @@ protected float computeWeight(long sequenceIndex) {
             randomNumber = random.nextDouble();
         }
 
-        return (float) (-sequenceIndex * lambda + Math.log(-Math.log(randomNumber)));
+        mostRecentTimeStamp = (mostRecentTimeStamp < sequenceIndex) ? sequenceIndex : mostRecentTimeStamp;
```

Comment:
minor. duplicated code.

Reply:
Yes. These two classes should be refactored out of a common base class. Issue 139.

Reply:
refactored.